server/application-server/src/main/java/de/tum/in/www1/hephaestus/leaderboard/LeaguePointsCalculationService.java

```diff
@@ -0,0 +1,119 @@
+package de.tum.in.www1.hephaestus.leaderboard;
+
+import de.tum.in.www1.hephaestus.gitprovider.pullrequest.PullRequest;
+import de.tum.in.www1.hephaestus.gitprovider.user.User;
+import java.time.OffsetDateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+@Service
+public class LeaguePointsCalculationService {
+    private final Logger logger = LoggerFactory.getLogger(LeaguePointsCalculationService.class);
+
+    // Starting points for new players
+    public static int POINTS_DEFAULT = 1000;
+    // Upper bound for first reducting in the k-factor
+    public static int POINTS_THRESHOLD_HIGH = 1750;
+    // Lower bound for first reducting in the k-factor
+    public static int POINTS_THRESHOLD_LOW = 1250;
+    // Minimum amount of points to decay each cycle
+    public static int DECAY_MINIMUM = 10;
+    // Factor to determine how much of the current points are decayed each cycle
+    public static double DECAY_FACTOR = 0.05;
+    // K-factors depending on the player's league points
+    public static double K_FACTOR_NEW_PLAYER = 2.0;
+    public static double K_FACTOR_LOW_POINTS = 1.5;
+    public static double K_FACTOR_MEDIUM_POINTS = 1.2;
+    public static double K_FACTOR_HIGH_POINTS = 1.1;
+
+
+
+    public int calculateNewPoints(User user, LeaderboardEntryDTO entry) {
+        // Initialize points for new players
+        if (user.getLeaguePoints() == 0) {
+            user.setLeaguePoints(POINTS_DEFAULT);
+        }
+
+        int oldPoints = user.getLeaguePoints();
+
+        double kFactor = getKFactor(user);
+        // Base decay
+        int decay = calculateDecay(user.getLeaguePoints());
+        // Bonus based on leaderboard score
+        int performanceBonus = calculatePerformanceBonus(entry.score());
+        // Additional bonus for placements
+        int placementBonus = calculatePlacementBonus(entry.rank());
+        // Calculate final point change
+        int pointChange = (int) (kFactor * (performanceBonus + placementBonus - decay));
+        // Apply minimum change to prevent extreme swings
+        int newPoints = Math.max(0, oldPoints + pointChange);
```

Comment:
The min points should not be `0` otherwise it will be reset to `POINTS_DEFAULT=1000`.

```suggestion
        int newPoints = Math.max(1, oldPoints + pointChange);
```